pallets/passkey/src/lib.rs

```diff
@@ -350,3 +365,65 @@ where
 		)
 	}
 }
+
+/// Block resource (weight) limit check.
+#[derive(Encode, Decode, Clone, TypeInfo)]
+#[scale_info(skip_type_params(T))]
+pub struct PasskeyWeightCheck<T: Config + Sync + Send>(pub PasskeyPayload<T>);
+
+impl<T: Config> PasskeyWeightCheck<T>
+where
+	T: Sync + Send,
+	<T as frame_system::Config>::RuntimeCall:
+		From<Call<T>> + Dispatchable<Info = DispatchInfo, PostInfo = PostDispatchInfo>,
+{
+	/// creating a new instance
+	pub fn new(passkey_payload: PasskeyPayload<T>) -> Self {
+		Self(passkey_payload)
+	}
+
+	/// Validate the transaction
+	pub fn validate(&self) -> TransactionValidity {
+		let some_call: &<T as Config>::RuntimeCall = &self.0.passkey_call.call;
+		let info = &some_call.get_dispatch_info();
+		let overhead = T::WeightInfo::pre_dispatch();
+		let final_info = DispatchInfo {
+			weight: overhead.saturating_add(info.weight),
+			class: info.class,
+			pays_fee: Pays::Yes, // ?
```

Comment:
Should this get hardcoded to always paying or it would better to get it from the dispatch_info?

Reply:
We should get this from call.get_dispatch_info ?

Reply:
Perhaps, I am missing something here but it does seem like we can get this from dispatch_info()?